pkg/fluxApplication/FluxApplicationService.go

```diff
@@ -0,0 +1,126 @@
+package fluxApplication
+
+import (
+	"context"
+	"github.com/devtron-labs/common-lib/utils/k8s"
+	"github.com/devtron-labs/devtron/api/helm-app/gRPC"
+	"github.com/devtron-labs/devtron/pkg/cluster"
+	"github.com/devtron-labs/devtron/pkg/fluxApplication/bean"
+	"go.uber.org/zap"
+)
+
+type FluxApplicationService interface {
+	ListApplications(ctx context.Context, clusterIds []int) ([]*bean.FluxApplicationListDto, error)
+	ConvertClusterBeanToClusterConfig(clusters []*cluster.ClusterBean) ([]*gRPC.ClusterConfig, error)
+}
+
+type FluxApplicationServiceImpl struct {
+	logger *zap.SugaredLogger
+	//helmAppService service.HelmAppService
+	clusterService cluster.ClusterService
+	helmAppClient  gRPC.HelmAppClient
+}
+
+func NewFluxApplicationServiceImpl(logger *zap.SugaredLogger,
+	//helmAppService service.HelmAppService,
+	clusterService cluster.ClusterService,
+	helmAppClient gRPC.HelmAppClient) *FluxApplicationServiceImpl {
+	return &FluxApplicationServiceImpl{
+		logger: logger,
+		//helmAppService: helmAppService,
+		clusterService: clusterService,
+		helmAppClient:  helmAppClient,
+	}
+
+}
+
+func (impl *FluxApplicationServiceImpl) ListApplications(ctx context.Context, clusterIds []int) ([]*bean.FluxApplicationListDto, error) {
+	var clusters []*cluster.ClusterBean
+	var err error
+	appListCluster := make([]*bean.FluxApplicationListDto, 0)
+	req := &gRPC.AppListRequest{}
+	if len(clusterIds) > 0 {
+		//for _, clusterId := range clusterIds {
+		//	clusterConfig, err := impl.helmAppService.GetClusterConf(clusterId)
+		//	if err != nil {
+		//		impl.logger.Errorw("error in getting clusters by ids", "err", err, "clusterIds", clusterIds)
+		//		return nil, err
+		//	}
+		//	req.Clusters = append(req.Clusters, clusterConfig)
+		//}
+
+	} else {
+		clusters, err = impl.clusterService.FindAll()
+		if err != nil {
+			impl.logger.Errorw("error in getting all active clusters", "err", err)
+			return nil, err
+		}
+
+		configs, err1 := impl.ConvertClusterBeanToClusterConfig(clusters)
+		if err1 != nil {
+			impl.logger.Errorw("error in getting all active clusters", "err", err1)
+			return nil, err1
+		}
+		req.Clusters = configs
+
+	}
+
+	//fluxAppsClusterCount := make(map[int32]int)
+
+	applicationStream, err := impl.helmAppClient.ListFluxApplication(ctx, req)
+	if err == nil {
+		fluxApplicationList, err1 := applicationStream.Recv()
```

Comment:
no need to assign err as "err1" just use "err" only , reassigning err will anyway make sure that new error is in err variable